docs/pipelines/security/machine-users.mdx

```diff
@@ -17,31 +17,90 @@ This guide will take approximately 30 minutes to complete.
-
-## Storing secrets
+## Storing Secrets
 
 During this setup, you will need to generate and securely store three GitHub tokens for two GitHub users. You will need a temporary location for these sensitive values between generating them and storing them in GitHub Actions. Do so according to your company's recommended security best practices (e.g., do not store them in Slack, a sticky note, etc., during this exercise.)
 
-
 :::note
 Your organization is required to rotate the GitHub tokens and update all GitHub secrets that use them.
 :::
 
 :::tip
 Gruntwork recommends that you use a password manager like [1Password](https://1password.com) to store these secrets securely.
+:::
 
+:::caution
 If you are screen sharing when generating these tokens, **hide or pause your screen** before selecting the `Generate token` button to prevent the token from being exposed.
 :::
 
+## Generating Tokens
+
+GitHub has two types of tokens that can be generated:
+
+1. Classic tokens
+2. Fine-grained tokens
+
+### Classic Tokens
+
+Classic tokens have coarse granularity and for the most part grant similar access to the user that generated them.
+
+The restrictions that can be applied to them are few, but they are still useful for some limited use cases.
+
+In Pipelines, the `ci-read-only-user` will need a Classic token because it needs to access software across **multiple GitHub organizations**. This is something that classic tokens can do that fine-grained tokens cannot.
+
+:::tip
+The `PIPELINES_READ_TOKEN` token must be created as a **Classic token**.
+:::
+
+### Fine-grained
+
+Fine-grained tokens have more granular permissions, and are generally recommended over Classic tokens when they are applicable.
+
+The only limitations on fine-grained tokens that are relevant from a Pipelines perspective is that:
+
+1. They have a single **Resource owner**, which is the organization that is allowed to access the token.
+2. They must have an expiration shorter than 1 year (Classic tokens can be configured to never expire).
+
+In Pipelines, the `ci-user` will need two fine-grained tokens because they allow for very limited access to specific repositories, with minimal permissions to them.
+
+:::tip
+The `INFRA_ROOT_WRITE_TOKEN` and `ORG_REPO_ADMIN_TOKEN` tokens must be created as **fine-grained tokens**.
+:::
+
+:::tip
+The **Resource owner** must be set to the GitHub organization of the repositories that the tokens will be used to access (e.g. `infrastructure-live-root`).
+:::
+
+![Resource owner](/img/pipelines/security/resource_owner.png)
+
+:::tip
+You might see your token status as **pending** after generating it. This is because some GitHub organizations require that fine-grained tokens be approved by an organization owner before they can be used.
+
+If you have the appropriate access, you can approve your new token by navigating to the settings of your GitHub organization and finding the token in the **Pending requests** section.
```

Comment:
this took me ages to find.
FYI the docs are [here](https://docs.github.com/en/organizations/managing-programmatic-access-to-your-organization/managing-requests-for-personal-access-tokens-in-your-organization)

Reply:
Thanks for the tip!